### ISSUE TITLE
Setting mouse mode to Input.MOUSE_MODE_CONFINED hides the cursor

### ISSUE THREAD

**Issue by jamestrotter** [godotengine/godot#19412]
**Godot version:**
3.0.3 RC3

**OS/device including version:**
Windows 10, Intel(R) HD Graphics 620

**Issue description:**
Setting `Input.set_mouse_mode` to `Input.MOUSE_MODE_CONFINED` hides the cursor. The documentation at http://docs.godotengine.org/uk/latest/classes/class_input.html#enum-input-mousemode says "[Confined mouse mode] Makes the mouse cursor visible but confines it to the game window."

Full disclosure I initially saw this behaviour with a C# script and assumed it was a C# issue, but I also tested it with GD script and the same issue occurred. If you alt-tab away from the window and back to it the cursor then shows as expected and is confined to the window

**Steps to reproduce:**
Call `Input.set_mouse_mode(Input.MOUSE_MODE_CONFINED)`

**Minimal reproduction project:**

[test.zip](https://github.com/godotengine/godot/files/2080737/test.zip)

**Comment by guilhermefelipecgs** [comment]
@jamestrotter You're using mono version? I can't reproduce this with 3.0.3 RC3.

**Comment by jamestrotter** [comment]
Weird. Maybe its my machine. Mono 5.12.0, but using the MSBuild build tool. I assume mono wouldn't affect the gdscript version also not working though?

**Comment by gswashburn** [comment]
I have tested this too and am looking for an answer. I have a HP touchscreen laptop with Intel 4600 GPU. The mouse cursor disappears when confined (GDScript). Which is useful for my game, but when I set mouse back to visible it stays hidden till I move outside of Viewport. It then becomes visible again. I am going to try on different dev machine with Nvidia card and see if same happens.
 I am using Godot 3.05, with GDScript.

**Comment by gswashburn** [comment]
I have tested on another machine with a NVidia 1050 GPU instead of Intel Graphics with same results. Mouse disappears when confined. When set back to visible you must move outside of viewport to get mouse cursor back.

**Comment by Jcozman** [comment]
I am having the same issue on mono 3.0.5. Setting the mouse mode to confined will hide the mouse. I have an option button with various windowed options, and I am setting fullscreen modes and mouse input like so:

```
private void OnOptionItemSelected(int i)
        {
            switch (i)
            {
                case 0:
                    OS.SetWindowFullscreen(true);
                    Input.SetMouseMode(Input.MouseMode.Confined);
                    break;
                case 1:
                    Input.SetMouseMode(Input.MouseMode.Visible);
                    OS.SetWindowFullscreen(true);
                    OS.SetBorderlessWindow(true);
                    break;
                case 2:
                    Input.SetMouseMode(Input.MouseMode.Visible);
                    OS.SetWindowFullscreen(false);
                    OS.SetBorderlessWindow(false);
                    break;
                default:
                    Input.SetMouseMode(Input.MouseMode.Visible);
                    OS.SetWindowFullscreen(false);
                    OS.SetBorderlessWindow(false);
                    break;
            }
        }
```

**Comment by wozeparrot** [comment]
Same issue here. Running Godot 3.0.5 not the mono version.

**Comment by TheBlue-1** [comment]
I have the same Issue in newest Mono version, but the cursor just stays invisible. If i set the cursor to captured and then to confined without moving the cursor intermediately it stays invisible. If i move it before the second change it gets visible.

**Comment by TheMoye** [comment]
Hi everyone,

Same issue for me : when I switch the mouse capture mode on Windows from **Captured** to **Confined**, the cursor remains invisible. But seems to move correctly in the window (I have a raycast feedback in the console, which tells me when I hover over an object).
Under Linux no problem.

I use Godot in version 3.2.1 Stable and 3.2.2 Alpha.

**Comment by Chaosus** [comment]
This seems fixed in 3.4/4.0.

**Comment by Calinou** [comment]
Note that in the `master` branch, `Input.MOUSE_MODE_CONFINED_HIDDEN` is available. It's not backported to `3.x` yet though.
